docs/src/design/decoder/main.md

```diff
@@ -355,7 +361,8 @@ In the above diagram, `blk` is the ID of the block which is about to finish exec
 
 When the VM executes an `END` operation, it does the following:
 
-1. Removes a tuple `(blk, prnt, f1)` from the block stack table.
+1. Removes a tuple `(blk, prnt, f1, ctx_next, fmp_next, b0_next, b1_next, fn_hash_next[0..4])` from the block stack table.
+    - in the above, the `x_next` variables denote the column `x` in the next row
```

Comment:
One thing that is not clear to me is how we can differentiate between `CALL`/`SYSCALL` and other blocks here.

For example, referring to `fmp_next` would work only if we are ending `CALL`/`SYSCALL` block - right? If we are ending from something like a `JOIN` block, `fmp_next` may not be `0` and so the row value won't match. Or am I missing something?

Reply:
Actually, the way I think this is maybe handled is via `IS_CALL` and `IS_SYSCALL` flags that we don't have documented here (these would be similar to `IS_LOOP_BODY` and `IS_LOOP` flags we denote as `f0` and `f1` in the diagram.

I don't remember too well now, but I think the `IS_CALL` and `IS_SYSCALL` flags were meant to be used for computing the proper value of the row to be removed from the block stack table. We should check the constraints for this and see how their degree would be affected once these flags are taken into account.

Reply:
Oops, right, I forgot to document that. It is now fixed - I also updated the block stack table constraints (which I had also forgotten).

> We should check the constraints for this and see how their degree would be affected once these flags are taken into account.

$u_{end}$ went from degree 5 to degree 6, so nothing changes